targets/unity-v2/source/Authentication/PlayFabAuthService.cs

```diff
@@ -0,0 +1,268 @@
+        /// Incompatible with split-screen/multi-instance play (issue to be fixed).
+        /// </summary>
+        public string RememberMeId
+        {
+            get { return PlayerPrefs.GetString(_PlayFabRememberMeIdKey, ""); }
+            set
+            {
+                var guid = string.IsNullOrEmpty(value) ? Guid.NewGuid().ToString() : value;
+                PlayerPrefs.SetString(_PlayFabRememberMeIdKey, guid);
+            }
+        }
+
+        /// <summary>
+        /// Check PlayerPrefs, and create new remember me id if empty.
+        /// </summary>
+        public string GetOrCreateRememberMeId()
+        {
+            var rememberMeId = PlayerPrefs.GetString(_PlayFabRememberMeIdKey, "");
+
+            if (string.IsNullOrEmpty(rememberMeId))
+            {
+                rememberMeId = Guid.NewGuid().ToString();
+                PlayerPrefs.SetString(_PlayFabRememberMeIdKey, rememberMeId);
+            }
+
+            return rememberMeId;
+        }
+
+        private readonly Dictionary<AuthTypes, IAuthenticationStrategy> _authStrategies = new Dictionary<AuthTypes, IAuthenticationStrategy>();
+
+        public PlayFabAuthService()
+        {
+            var strategyTypes = typeof(IAuthenticationStrategy).Assembly.GetTypes().Where(type => typeof(IAuthenticationStrategy).IsAssignableFrom(type) && !type.IsInterface);
+
+            foreach (var strategyType in strategyTypes)
+            {
+                var strategy = (IAuthenticationStrategy) Activator.CreateInstance(strategyType);
+                _authStrategies.Add(strategy.AuthType, strategy);
+            }
+        }
+
+        public bool IsClientLoggedIn()
+        {
+            return AuthenticationContext != null && AuthenticationContext.IsClientLoggedIn();
+        }
+
+        public bool IsEntityLoggedIn()
+        {
+            return AuthenticationContext != null && AuthenticationContext.IsEntityLoggedIn();
+        }
+
+        public void ClearRememberMe()
+        {
+            PlayerPrefs.DeleteKey(_LoginRememberKey);
+            PlayerPrefs.DeleteKey(_PlayFabRememberMeIdKey);
+        }
+
+        /// <summary>
+        /// Kick off the authentication process by specific authType.
+        /// </summary>
+        /// <param name="authType"></param>
+        /// <param name="authKeys"></param>
+        public void Authenticate(AuthTypes authType, AuthKeys authKeys = null)
+        {
+            AuthType = authType;
+            Authenticate(authKeys);
+        }
+
+        public void Authenticate(AuthKeys authKeys = null)
+        {
+            if (AuthType == AuthTypes.None)
+            {
+                if (OnDisplayAuthentication != null)
+                    OnDisplayAuthentication.Invoke();
+
+                return;
+            }
+
+            if (RememberMe && !string.IsNullOrEmpty(RememberMeId))
+            {
+                PlayFabClientAPI.LoginWithCustomID(new LoginWithCustomIDRequest
```

Comment:
I am not sure why we are doing this.  I do understand the scenario where a user logs in with a email & password and we have to hash that into a remember me id,  but this insinuates that under all conditions when you use the remember me feature that it is a login with a custom id, which is a little misleading in the "auto-login" as it remembers your previous auth type.   This issue isn't a blocker for release but we should revisit it from a design perspective.